src/stdlib/io/file.rs

```diff
@@ -0,0 +1,323 @@
+    read_position: RefCell<usize>,
+}
+
+impl IoFile {
+    pub fn new(file: File) -> Self {
+        Self {
+            inner: Rc::new(Either::Left(RefCell::new(Some(file)))),
+            read_position: RefCell::new(0),
+        }
+    }
+    pub fn stdin() -> Self {
+        Self {
+            inner: Rc::new(Either::Right(StdFileKind::Stdin)),
+            read_position: RefCell::new(0),
+        }
+    }
+    pub fn stdout() -> Self {
+        Self {
+            inner: Rc::new(Either::Right(StdFileKind::Stdout)),
+            read_position: RefCell::new(0),
+        }
+    }
+    pub fn stderr() -> Self {
+        Self {
+            inner: Rc::new(Either::Right(StdFileKind::Stderr)),
+            read_position: RefCell::new(0),
+        }
+    }
+    pub fn is_std(&self) -> bool {
+        matches!(self.inner.as_ref(), Either::Right(_))
+    }
+    pub fn close(&self) -> Result<(), io::Error> {
+        if self.is_std() {
+            Ok(())
+        } else {
+            let Either::Left(left) = self.inner.as_ref() else {
+                unreachable!()
+            };
+
+            let mut file_lock = left.borrow_mut();
+            if let Some(file_handle) = file_lock.take() {
+                file_handle.sync_all()?;
+            }
+            Ok(())
+        }
+    }
+    pub fn flush(&self) -> Result<(), io::Error> {
+        match self.inner() {
+            Either::Left(left) => {
+                let mut file_lock = left.borrow_mut();
+                if let Some(ref mut file_handle) = *file_lock {
+                    file_handle.flush()?;
+                }
+                Ok(())
+            }
+            Either::Right(kind) => {
+                if matches!(kind, StdFileKind::Stdout | StdFileKind::Stderr) {
+                    io::stdout().flush()
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    }
+    pub fn read_with_format<'gc>(
+        &self,
+        ctx: Context<'gc>,
+        format: &str,
+    ) -> Result<Option<Value<'gc>>, Error<'gc>> {
+        match self.inner() {
+            Either::Left(left) => {
+                let mut file = left.borrow_mut();
+                let file = match *file {
+                    Some(ref mut file) => file,
+                    None => return Err("attempt to use a closed file".into_value(ctx).into()),
+                };
+                let mut read_state = self.read_position.borrow_mut();
+                read_from_any(ctx, file, format, &mut read_state)
+            }
+            Either::Right(kind) => match kind {
+                StdFileKind::Stdin => {
+                    let mut read_bytes = self.read_position.borrow_mut();
+                    let mut stdin = io::stdin();
+                    let mut buf = Vec::new();
+                    stdin.read_exact(&mut buf)?;
+                    let mut cursor = Cursor::new(buf);
+                    read_from_any(ctx, &mut cursor, format, &mut read_bytes)
```

Comment:
This approach won't work, since always reads a fixed amount of data (currently 0 bytes).  I assume this was a work-around for stdin/stdout not supporting Seek, but `read_from_any` just needs to be changed to support non-seekable files.